File: analyzers/tests/SonarAnalyzer.Rules.Tests/Cases/FieldShouldNotBePublic.vb
Namespace Tests.TestCases
    Public Class SomeClass
        Public Shared Pi As Double = 3.14
        Public Const Pi2 As Double = 3.14
        Public ReadOnly Pi3 As Double = 3.14 ' Noncompliant {{Make 'Pi3' private.}}
'                       ^^^
        Public Pi4 As Double = 3.14 ' Noncompliant
        Private Pi5 As Double = 3.14
        Protected Pi6 As Double = 3.14
        Friend Pi7 As Double = 3.14

        Public Class MyPublicSubClass

            Public Pi4 As Double = 3.14 ' Noncompliant
            Private Pi5 As Double = 3.14
            Protected Pi6 As Double = 3.14
            Friend Pi7 As Double = 3.14
        End Class

        Private Class MyPrivateSubClass

            Public Pi4 As Double = 3.14
            Private Pi5 As Double = 3.14
            Protected Pi6 As Double = 3.14
            Friend Pi7 As Double = 3.14
        End Class

        Protected Class MyProtectedSubClass

            Public Pi4 As Double = 3.14 ' Noncompliant
            Private Pi5 As Double = 3.14
            Protected Pi6 As Double = 3.14
            Friend Pi7 As Double = 3.14
        End Class

        Friend Class MyInternalSubClass

            Public Pi4 As Double = 3.14
            Private Pi5 As Double = 3.14
            Protected Pi6 As Double = 3.14
            Friend Pi7 As Double = 3.14
        End Class
    End Class

    Public Structure MyStruct
        Public Shared Pi As Double = 3.14
        Public Const Pi2 As Double = 3.14

        Public Pi3 As Double
        Private Pi4 As Double
        Friend Pi5 As Double

        Public Structure MyPublicSubStruct
            Public Pi4 As Double
            Private Pi5 As Double
            Friend Pi7 As Double
        End Structure

        Private Structure MyPrivateSubStruct
            Public Pi4 As Double
            Private Pi5 As Double
            Friend Pi7 As Double
        End Structure

        Friend Structure MyInternalSubStruct
            Public Pi4 As Double
            Private Pi5 As Double
            Friend Pi7 As Double
        End Structure
    End Structure
End Namespace

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldNotBePublic.cs
namespace Tests.TestCases
{
    public class MyClass
    {
        public static double Pi = 3.14;
        public const double Pi2 = 3.14;
        public readonly double Pi3 = 3.14; // Noncompliant {{Make 'Pi3' private.}}
//                             ^^^
        public double Pi4 = 3.14; // Noncompliant
        private double Pi5 = 3.14;
        protected double Pi6 = 3.14;
        internal double Pi7 = 3.14;

        public class MyPublicSubClass
        {
            public double Pi4 = 3.14; // Noncompliant
            private double Pi5 = 3.14;
            protected double Pi6 = 3.14;
            internal double Pi7 = 3.14;
        }

        private class MyPrivateSubClass
        {
            public double Pi4 = 3.14;
            private double Pi5 = 3.14;
            protected double Pi6 = 3.14;
            internal double Pi7 = 3.14;
        }

        protected class MyProtectedSubClass
        {
            public double Pi4 = 3.14;  // Noncompliant
            private double Pi5 = 3.14;
            protected double Pi6 = 3.14;
            internal double Pi7 = 3.14;
        }

        internal class MyInternalSubClass
        {
            public double Pi4 = 3.14;
            private double Pi5 = 3.14;
            protected double Pi6 = 3.14;
            internal double Pi7 = 3.14;
        }
    }

    public struct MyStruct
    {
        public static double Pi = 3.14;
        public const double Pi2 = 3.14;

        public double Pi3;
        private double Pi4;
        internal double Pi5;

        public struct MyPublicSubStruct
        {
            public double Pi4;
            private double Pi5;
            internal double Pi7;
        }

        private struct MyPrivateSubStruct
        {
            public double Pi4;
            private double Pi5;
            internal double Pi7;
        }

        internal struct MyInternalSubStruct
        {
            public double Pi4;
            private double Pi5;
            internal double Pi7;
        }
    }
}